eecs280staff/tutorials
# IDE Tutorial Overhaul

Move project creation from the main setup tutorial to the IDE tutorials.  As a result, both the main setup tutorial and the IDE tutorials are simpler.

This PR includes a re-write of each IDE tutorial.  It also reduces the p1-specific vibes of the IDE tutorials.

### Changes
- Move "Create a project folder" step from main tutorial to IDE tutorials
- Move "Download starter files" step from main tutorial to IDE tutorials
- Generic `main.cpp` is generic, not specific to EECS 280 P1
- Provide sample code in Debug section, replacing example that depended on the EECS 280 project 1 file structure.
- Code goes in `src/` subdirectory

Lots of VS Code changes
- Do not provide sample `launch.json`.  The default works fine nowadays.
- Prefer to install extensions using `code --install-extension`, remove a few screenshots.
- Remove CodeLLDB extension now that Apple Silicon is supported by the official Microsoft C/C++ extension
- Remove Intellisense configuration for C+11.  It looks like C++17 is the default now.  Replace with a note in trouble shooting.
- Remove CAEN Linux special version loading.  A recent version is now the default.

### Prerequisites
- #29

### Validation
Go through the entire tutorial and make sure it works on different platforms.
- [ ] macOS/Intel
- [ ] macOS/Apple Silicon
- [ ] Windows WSL
- [ ] CAEN Linux

### Related issues
- Closes #14
- Closes #16
- Closes #25
- Closes #28
- Helps #10
- Helps #15

### After merging
- Update P1 spec
  - `stats.cpp` starter code with function stubs and reminder to configure target
  - `main.cpp` starter code and reminder to configure target
  - `tree` command to verify files are in the right place
- Update P2-P5 specs
  - Replace Setup section with:
    1. Group registration
    2. Link to IDE project setup tutorials
       - Provide starter files link, suggested project folder name
    3. Link to version control tutorial
  - Add reminder to each major code section to update the IDE build target
  - `tree` command to verify files are in the right place
